lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -822,7 +822,7 @@ public function count($criteria = array())
             ? $this->expandCriteriaParameters($criteria)
             : $this->expandParameters($criteria);
 
-        return $this->conn->executeQuery($sql, $params, $types)->fetchColumn();
+        return (int) $this->conn->executeQuery($sql, $params, $types)->fetchColumn();
```

Comment:
Just checked: DBAL forces PDO to throw exceptions, so this `fetchColumn` is actually OK, no changes needed to your PR :-)